### ISSUE TITLE
Bug: sort-imports - Dependencies with typescript import-equals not detected

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#513]
# Describe the bug

In typescript projects targetting Node.js, typescript import-equals re-order may result in runtime errors due to usage before initialization.

## Code example

Take the following typescript configuration:

```json
{
  "compilerOptions": {
    "moduleResolution": "NodeNext",
    "module": "nodenext"
  }
}
```

Code

```ts
// errors-namespace.ts

export namespace ErrorsNamespace {

  export const ERROR = 'error';

}
```

```ts
// main.ts

// Create a config to get this order
import MainError = ErrorsNamespace.ERROR;
import { ErrorsNamespace } from "./errors-namespace";
```

Compilation result:

```ts
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var MainError = errors_namespace_1.ErrorsNamespace.ERROR;  // <-- Will throw "Cannot access 'errors_namespace_1' before initialization
"
const errors_namespace_1 = require("./errors-namespace");
console.log(MainError);
```

### `default` import

```ts
import a = aImport.a1.a2;
import aImport from "b";
```

### `wildcard` import

```ts
import a = aImport.a1.a2;
import * as aImport from "b";
```

### `named` import

```ts
import a = aImport.a1.a2;
import { aImport } from "b";
```

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

4.11.0 but exists in lower versions as well

### Validations

- [x] Read the [docs](https://perfectionist.dev).
- [x] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.
